test/e2e/adminactions_redeployvm.go

```diff
@@ -0,0 +1,117 @@
+package e2e
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	mgmtcompute "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-03-01/compute"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func newVirtualMachinesClient(subscriptionID string) (*mgmtcompute.VirtualMachinesClient, error) {
+	client := mgmtcompute.NewVirtualMachinesClient(os.Getenv("AZURE_SUBSCRIPTION_ID"))
```

Comment:
I'd use the client here: https://github.com/Azure/ARO-RP/blob/master/pkg/util/azureclient/mgmt/compute/virtualmachines.go
you could also add the list functionality into the _addons.go
https://github.com/Azure/ARO-RP/blob/master/pkg/util/azureclient/mgmt/compute/virtualmachines_addons.go